docs/operate/reference/advanced-modules/_index.md

```diff
@@ -46,6 +46,74 @@ Running {{< glossary_tooltip term_id="modular-resource" text="modular resources"
 
 However, if you are unable to use modular resources because you need to host `viam-server` on a non-Linux system or have an issue with compilation, you may need to [implement a custom component and register it on a server configured as a remote](/operate/reference/advanced-modules/custom-components-remotes/) on your machine.
 
+## Package and deploy using Docker
+
+In rare cases, you may need to package and deploy a module using Docker.
+Use cases for this include:
+
+- Your module has complex system dependencies that cannot be easily installed on a machine.
+- You have a large bundle and want to use layer caching to reduce the size of the download.
+- You have specific security requirements that are difficult to meet with the default module deployment.
+
+If you choose to deploy your module using Docker, we recommend creating a "first run" script or binary to run any necessary setup steps.
+
+{{% expand "Click for first run script instructions" %}}
+
+1. Create a tarball that contains:
+
+   - The module's entrypoint script or binary
+   - A <file>meta.json</file>
+   - A first run script or binary that will be executed during the setup phase, for example:
+
+     ```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+     #!/usr/bin/env bash
+
+     if [[ -n "$VIAM_TEST_FAIL_RUN_FIRST" ]]; then
+         echo "Sorry, I've failed you."
+         exit 1
+     fi
+
+     docker pull mongo:6
+
+     cat << EOF
+     -------------------------------------
+     Congratulations!
+     The setup script ran successfully!
+     This message is obnoxiously large for
+     testing purposes.
+     Sincerely,
+     First Run Script
```

Comment:
Is it necessary to show a large message for the docs?

Reply:
I will reduce it. This was Max's original message and thought it might be okay to bring some amusement to the contents of an expander, further hidden within a hidden page, but no it doesn't need to be here.
```suggestion
     The setup script ran successfully!
```